_clanky/mapa-s-výsledky-voleb-po-okrscích.md

```diff
@@ -0,0 +1,115 @@
+    ndjson-split 'd.features' < volebni_okrsky-simple.json > volebni_okrsky-simple.ndjson
+    csv2json -n statistics.csv > statistics.ndjson
+    ndjson-join --left 'd.id' volebni_okrsky-simple.ndjson statistics.ndjson | ndjson-map 'Object.assign(d[0], Object.assign(d[0].properties, d[1]))' > volebni_okrsky-simple-data.ndjson
+    geo2topo tracts=volebni_okrsky-simple-data.json > volebni_okrsky-simple-data-topo.json
+
+Ještě si připravíme soubor s barvami politických stran, použijeme vlastní krátký skript `join_parties.py` a přidáme [barvy][link_political_parties].
+
+A nyní již jen vše použijeme do výsledného html souboru, kde mírně upravíme [index.html](https://data.gov.cz/attachments/%C4%8Dl%C3%A1nky/kartogram-choropleth/data/index.html) z [Kartogram ČR][link_choropleth]: Nový [index.html](../attachments/články/mapa-s-výsledky-voleb-po-okrscích/data/index.html).
+
+### Další užití
+Volební mapy lze použít prakticky pro libovolné volby, referenda, apod. 
+
+Detailní mapa např. umožňuje i pohled dovnitř měst, např. zde jsou zobrazeny výsledky krajských voleb 2020 v Plzni a nejbližším okolí, kde jsou zřetelně vidět ostrůvky - sídliště - v moři modré (ODS), kde vyhrávalo fialové ANO.
+
+{% include image.html url="../attachments/články/mapa-s-výsledky-voleb-po-okrscích/images/plzen_popisy.png" description="Plzeň a okolí - krajské volby 2020 po volebních okrscích: barva odpovídá vítězné straně v daném volebním okrsku: ANO (fialová), ODS + TOP 09 (modrá), STAN + SZ + PRO Plzeň (zelená), Piráti (černá). Je zde vidět, že ANO vyhrávalo hlavně na sídlištích." %}
+
+### Použité nástroje a zdroje:
+- [Kartogram ČR (choropleth, choropletová mapa)][link_choropleth]
+- [TopoJSON-simplify][link_simplify]
+- [NDJSON-CLI][link_cli]
+- [TopoJSON-client][link_client]
+- [TopoJSON-server][link_server]
+- [Streming Shapefile Parser][link_parser]
+- [D3.js][link_d3]
+
+### Závěr
+Článek ukazuje jednu z metod, jak jednoduše vykreslit mapu s volebními výsledky voleb na úrovni volebních okrsků. Zároveň demonstruje, jak používat volební otevřená data z ČSÚ a mapová otevřená data z RÚIAN.
+
+
+[link_rozhlas]: https://www.irozhlas.cz/volby/jak-volili-vasi-sousedi-prohlednete-si-nejpodrobnejsi-mapu-volebnich-vysledku_1710220940_pek "Jak volili vaši sousedi. Prohlédněte si nejpodrobnější mapu volebních výsledků."
+[link_lidovky]: https://www.lidovky.cz/prezidentske-volby-2018.aspx?k=vysledky-druhe-kolo&t=okrsky-mapa "Prezidentské volby 2018."
+[link_volebniatlas]: https://volebniatlas.cz/ "Volební atlas"
+[link_sulek]: https://marcel.sulek.eu/2018/02/05/volebni-mapy.html "Volební mapy"
+[link_choropleth_qgis]: https://data.gov.cz/%C4%8Dl%C3%A1nky/kartogram-choropleth#postup-zpracov%C3%A1n%C3%AD-mapa-v-qgis "Kartogram ČR (choropleth, choropletová mapa): Postup zpracování v QGIS"
+[link_leaflet]: https://leafletjs.com/examples/choropleth/ "Leaflet: Choropleth"
+[link_nkod]: https://data.gov.cz/datov%C3%A9-sady "NKOD"
```

Comment:
lépe neencodované, tedy https://data.gov.cz/datové-sady - ostatní podobně

Reply:
to je ta debilní čeština v url - se to musí opravovat napůl ručně, to je fakt pruda, uááááá

Reply:
a to ještě v tomhle https://data.gov.cz/datov%C3%A1-sada?iri=https%3A%2F%2Fdata.gov.cz%2Fzdroj%2Fdatov%C3%A9-sady%2Fhttp---atom.cuzk.cz-api-3-action-package_show-id-cz-00025712-cuzk_ruian-staty-shp_1
se ani nedá použít url decode, protože to odkóduje i ty lomítka apod., uááááá na druhou :-)